### PR TITLE
Float functions

### PR DESCRIPTION
**Pull request by yashladd**

### CODE REVIEW

File: src/fpcore/ast_methods/substitute.py
Comment (IanBriggs) [inline src/fpcore/ast_methods/substitute.py:1]: Please move this to it's own file in `ast_methods`

File: src/fpcore/ast_methods/to_libm_c.py [hunk 1]
```diff
@@ -30,21 +31,22 @@ def to_libm_c(self, *args, **kwargs):
...
     return self.source
 
 
 @add_method(Constant)
-def to_libm_c(self):
+def to_libm_c(self, numeric_type=fp64()):
     try:
-        return _CONST_MAPPING[self.source]
+        is_float = isinstance(numeric_type, fp32)
+        return _CONST_MAPPING[self.source] if not is_float else f"(float) {_CONST_MAPPING[self.source]}"
```
Comment (IanBriggs) [inline src/fpcore/ast_methods/to_libm_c.py:42]: Good solution to this issue!

File: src/lambdas/cody_waite.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def replace_lambda(self, search, replace):
         if self.mod_cases is not None:
             new_cases = dict()
             for k, body in self.cases.items():
-                new_cases[i] = body.replace_lambda(search, replace)
+                new_cases[k] = body.replace_lambda(search, replace)
```
Comment (IanBriggs) [inline src/lambdas/cody_waite.py:56]: Thanks for fixing my mistake

**Review by IanBriggs — CHANGES_REQUESTED** [review]
Overall looks good. 👍 

Please move the substitute float operation code to its own file before merge.

File: src/fpcore/ast_methods/substitute.py
Comment (yashladd) [reply src/fpcore/ast_methods/substitute.py:1]: Done

File: src/fpcore/ast_methods/to_libm_c.py (hunk 1 above)
Comment (yashladd) [reply src/fpcore/ast_methods/to_libm_c.py:42]: Thanks!

**Comment by yashladd** [comment]
Thank you! merging

**Pull request merged**